object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionVisitor.java

```diff
@@ -61,6 +96,81 @@ public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
     return super.visitMethodInvocation(node, p);
   }
 
+  @Override
+  public Void visitNewClass(NewClassTree node, Void p) {
+
+    if (!isAssignedToLocal(this.getCurrentPath())) {
```

Comment:
Please create a method that includes the shared logic from `visitNewClass` and `visitMethodInvocation` and call it in both places, to remove duplication